packages/api/src/account/service.ts

```diff
@@ -50,7 +60,72 @@ const getSignupDetails = async (
   return signupDetailResult.rows[0];
 };
 
+const leaveArchive = async ({
+  emailFromAuthToken,
+  userSubjectFromAuthToken,
+  archiveId,
+  ip,
+}: LeaveArchiveRequest): Promise<{
+  accountArchiveId: string;
+}> =>
+  db.transaction(async (transactionDb) => {
+    const accountArchiveResult =
+      await transactionDb.sql<GetAccountArchiveResult>(
+        "account.queries.get_account_archive",
+        {
+          archiveId,
+          email: emailFromAuthToken,
+        }
+      );
+
+    const accountArchive = accountArchiveResult.rows[0];
+
+    if (!accountArchive) {
+      throw new createError.NotFound(
+        `Unable to find relationship with archiveId ${archiveId}`
+      );
+    }
+
+    if (accountArchive.accessRole === ACCESS_ROLE.Owner) {
+      throw new createError.BadRequest(
+        "Cannot leave archive while owning it. Either pass ownership to another account or delete archive."
+      );
+    }
+
+    const deleteResult = await db.sql<{ accountArchiveId: string }>(
+      "account.queries.delete_account_archive",
+      {
+        archiveId,
+        email: emailFromAuthToken,
+      }
+    );
+
+    if (!deleteResult.rows[0]) {
+      throw new createError.InternalServerError(
+        "Unexpected result while performing DELETE on account archive relationship."
+      );
+    }
+
+    const eventData: CreateEventRequest = {
+      action: EVENT_ACTION.Update,
+      entity: EVENT_ENTITY.Account,
+      entityId: accountArchive.accountId.toString(),
+      ip,
+      version: 1,
+      body: {
+        archiveId,
+        accountArchiveId: accountArchive.accountArchiveId,
+      },
+      userSubjectFromAuthToken,
```

Comment:
I don't think this event would communicate to the future reviewer of the event logs that this was a deletion of an `account_archive`. The way I'd communicate that would be to add an `account_archive` `entity` type and a `deleted` action type and make this an `account_archive.deleted` event.

Reply:
i think this is a valid point. perhaps we can decide how to move forward during today's planning session?

Reply:
We discussed and will do this - @valentinvsp will open a PR in backend to add those new values to the database.